gravitee-am-gateway/gravitee-am-gateway-handler/gravitee-am-gateway-handler-oidc/src/main/java/io/gravitee/am/gateway/handler/oauth2/resources/handler/risk/RiskAssessmentHandler.java

```diff
@@ -86,21 +98,38 @@ public void handle(RoutingContext routingContext) {
         final Optional<io.gravitee.am.model.User> user = ofNullable(routingContext.user())
                 .map(u -> (User) u.getDelegate())
                 .map(User::getUser);
-        var riskAssessment = client.map(Client::getRiskAssessment).orElse(new RiskAssessmentSettings());
+        var riskAssessment = getRiskAssessmentSettings();
         if (client.isPresent() && user.isPresent() && riskAssessment.isEnabled()) {
             computeRiskAssessment(routingContext, client.get(), user.get(), riskAssessment);
         } else {
             routingContext.next();
         }
     }
 
+    @SuppressWarnings("all")
+    private RiskAssessmentSettings getRiskAssessmentSettings() {
+        var settings = new RiskAssessmentSettings();
+        assessments.forEach((assessmentType, setter)  -> {
+            var enabled = environment.getProperty(String.format("alerts.risk_assessment.settings.%s.enabled", assessmentType), Boolean.class, false);
+            var thresholds = environment.getProperty(String.format("alerts.risk_assessment.settings.%s.thresholds", assessmentType), Map.class, emptyMap());
+
+            AssessmentSettings assessmentSettings = new AssessmentSettings().setEnabled(enabled);
+            if(enabled) {
+                assessmentSettings.setThresholds((Map<Assessment, Double>) thresholds);
+                settings.setEnabled(true);
+            }
+            setter.accept(settings, assessmentSettings);
+        });
+        return settings;
+    }
+
```

Comment:
I am not very certain that the Alert risk Assessment has to be processed here.

`io.gravitee.am.gateway.handler.common.alert.AlertEventProcessor` contains everything necessary to process a user authentication event.

If the authentication successful, you can check it within the AlertEventProcessor
```
                .property(PROPERTY_AUTHENTICATION_STATUS, eventItem.type().name())
```

then we decorate the AlertEvent with the assessment message

Maybe there is some abstraction work to think about so all the utility methods in risk assessment are reusable

Reply:
Yes I would have thought at least `RiskAssessmentHandler#computeRiskAssessment` should be available to `AlertEventProcessor` as well.

Reply:
Which in turn depends on three other methods and even more fields (e..g EventBus).